### PR TITLE
Prefer #after? to numeric time comparisons

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

ActiveSupport adds DateAndTime::Calculations#after? (and before?) -
which clarifies intent (users newer than the relative time are
skipped) of the early returns.

## QA Instructions, Screenshots, Recordings

No changes expected.

### UI accessibility concerns?

None

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: refactor
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._
- [x] This change does not need to be communicated, and this is why not: refactor

### CODE REVIEW

File: app/services/broadcasts/welcome_notification/generator.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@ def send_discuss_and_ask_notification
       end
 
       def send_download_app_notification
-        return if user.created_at > 7.days.ago || received_notification?(download_app_broadcast)
+        return if user.created_at.after?(7.days.ago) || received_notification?(download_app_broadcast)
```
Comment (djuber) [inline app/services/broadcasts/welcome_notification/generator.rb:79]: https://github.com/forem/forem/pull/16052#discussion_r782413219 the surrounding context (when we call this generator from the worker) is there won't be users older than 7 days and this always returns early without sending a notification.

**Comment by jeremyf** [comment]
This is such a great refactor.  I always hate doing `>` comparison with dates.  `after?` makes things so much more legible.

**Review by jeremyf — APPROVED** [review]

File: app/services/broadcasts/welcome_notification/generator.rb (hunk 1 above)
Comment (djuber) [reply app/services/broadcasts/welcome_notification/generator.rb:79]: I have a pr to fix that.

**Review by juliannatetreault — APPROVED** [review]
This is so much more readable. 🤩

**Pull request merged**